### PR TITLE
1 image upload

### PR DESCRIPTION
**Pull request by jphilly11**
Created a form to upload an image which then hits the backend and is saved then sent back to frontend to be displayed.

### CODE REVIEW

**Review by spanjshiri — CHANGES_REQUESTED** [review]
There are some merge conflicts that need to be addressed before merging. But otherwise this looks good.

File: ui/src/ImageUpload.tsx [hunk 1]
```diff
@@ -1,23 +1,33 @@
 import * as React from 'react';
+import axios from 'axios';
 
-class ImageUpload extends React.Component {
+class ImageUpload extends React.Component<any, any> {
```
Comment (KatherynRizek) [inline ui/src/ImageUpload.tsx]: Is adding the `<any, any>` for Typescript?

Comment (jphilly11) [reply ui/src/ImageUpload.tsx]: yes I added this for the typescript

File: api/src/main/java/com/exelaration/abstractmemery/controllers/UploadController.java [hunk 2]
```diff
@@ -1,36 +1,64 @@
...
         logger.info("\nGet Request\nTop Text: " + caption.getTopText() + "\nBottom Text: " + caption.getBottomText());
         return new ResponseEntity<>(caption, HttpStatus.OK);
-    }
-
-    @PostMapping("/")
-    public ResponseEntity newCaption(@RequestBody Caption caption) {
-        Logger logger = LoggerFactory.getLogger(UploadController.class);
+	}
+	
+	@PostMapping("/upload")
```
Comment (mchughmk) [inline api/src/main/java/com/exelaration/abstractmemery/controllers/UploadController.java]: The `@PostMapping` needs to be just "/"

File: ui/src/ImageUpload.tsx (hunk 1 above)
Comment (mchughmk) [reply ui/src/ImageUpload.tsx]: @kosmond13 For reference, this is telling Typescript the data types to use for the Props and State, respectively.

@jphilly11 We should refactor the state to be using a Typescript type.  I'll provide an example.

File: ui/src/ImageUpload.tsx [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
         e.preventDefault();
-        const data = new FormData();
-        data.append('image', e.target.file);
-        fetch('', {
-            method: 'POST',
-            body: data,
-        });
+        const formData = new FormData();
+        formData.append('file',this.state.file);
+        axios.post('http://localhost:8080/upload', formData)
```
Comment (mchughmk) [inline ui/src/ImageUpload.tsx]: The URL will need to end in "/", or else you'll get a CORS error.

**Review by mchughmk — CHANGES_REQUESTED** [review]

**Review by spanjshiri — DISMISSED** [review]

**Review by mchughmk — APPROVED** [review]

File: ui/src/App.tsx [hunk 4]
```diff
@@ -2,13 +2,15 @@ import React from 'react';
...
 import TextUpload from './TextUpload';
+import ImageUpload from './ImageUpload';
 
 const App = () => {
   return (
     <div className="App">
       <header className="App-header">
         <img src={logo} className="App-logo" alt="logo" />
         <TextUpload />
+        <ImageUpload/>
```
Comment (spanjshiri) [inline ui/src/App.tsx:13]: I think we need to eventually work on better styling than what we have, but that is for another time.

**Review by spanjshiri — APPROVED** [review]

**Pull request merged**